### PR TITLE
[Mapping] construct total mortar matrix with solver

### PR DESCRIPTION
**Pull request by peterjwilson**
**Description**
Adds option to construct the total mortar matrix with a solver.

### CODE REVIEW

**Review by philbucher — DISMISSED** [review]
awesome PR, super clean :)

**Comment by peterjwilson** [comment]
@philbucher although it's a little bit less clean than your last review, it's now ready for you to have a look at 😄 .

- 4 small tests in the mapping app (map, inverse map, dual mortar, precompute_mapping_matrix)
- EnforceConsistencyWithScaling brought back
- implement mapping matrix construction with solvers with parameters setting 'is_precompute_mapping_matrix'
- refactored MappingMatrixUtilities::CheckRowSum to use it as well
- added access method to get mapping matrix only if 'is_precompute_mapping_matrix' or is dual mortar

**Comment by philbucher** [comment]
awesome, thx a lot!
I will take a look asap and fix some things myself

File: applications/MappingApplication/custom_mappers/coupling_geometry_mapper.cpp [hunk 1]
```diff
@@ -285,61 +286,58 @@ void CouplingGeometryMapper<TSparseSpace, TDenseSpace>::EnforceConsistencyWithSc
...
+    #pragma omp parallel
+    {
+        Vector solution(n_rows);
+        Vector projector_column(n_rows);
+
+        #pragma omp for
+        for (int i = 0; i < static_cast<int>(mpMappingMatrix->size2()); ++i)
+        {
+            for (size_t j = 0; j < n_rows; ++j) projector_column[j] = rProjectedInterfaceMatrix(j, i); // TODO try boost slice or project
+            mpLinearSolver->Solve(rConsistentInterfaceMatrix, solution, projector_column);
```
Comment (philbucher) [inline applications/MappingApplication/custom_mappers/coupling_geometry_mapper.cpp:335]: @peterjwilson I am not sure if it is this is threadsafe to do in parallel, I am pretty sure it is not, depending on the linear solver used

**Review by philbucher — APPROVED** [review]
cool, I only made some minor modifications

see my comment regarding the thread-safety of the linear solution

**Pull request merged**